locallib.php

```diff
@@ -1268,18 +1283,24 @@ function moodleoverflow_print_post($post, $discussion, $moodleoverflow, $cm, $co
     $mustachedata->subject = format_string($post->subject);
 
     // Post was anonymized.
-    if ($post->userid != 0) {
+    if (!anonymous::is_post_anonymous($post, $moodleoverflow, $post->userid)) {
         // User picture.
         $mustachedata->picture = $OUTPUT->user_picture($postinguser, ['courseid' => $course->id]);
     }
 
     // The rating of the user.
-    $postuserrating = \mod_moodleoverflow\ratings::moodleoverflow_get_reputation($moodleoverflow->id, $postinguser->id);
+    if (anonymous::is_post_anonymous($post, $moodleoverflow, $post->userid) && $post->userid !== $USER->id) {
+        $postuserrating = '--';
```

Comment:
It would be great if in a completely anonymous instance, the reputation is not shown at all even for oneself as this value is not used/ updated.

Reply:
I have removed the display completely when anonymous and also cleaned up the code a bit :)